webapp/graphite/functions/params.py

```diff
@@ -135,11 +136,12 @@ def validateParams(func, params, args, kwargs):
       # requirement is satisfied from "args"
       value = args[i]
 
-    # parameter is restricted to a defined set of values, but value is not in it
+    # parameter is restricted to a defined set of possible values, but given value is not in it.
+    # possibly it's a deprecated but still accepted value.
     if params[i].options and value not in params[i].options:
-      raise InputParameterError(
-        'Invalid option specified for function "{func}" parameter "{param}"'.format(
-          param=params[i].name, func=func))
+      log.warning(
+        'given option "{option}" in function "{func}" is not in list of valid options, it may be deprecated but still accepted'
```

Comment:
I'd still include the parameter name here, and shorten up the text:
```
'Deprecated value "{value}" specified for parameter "{param}" of function "{func}"'
```
Longer-term we pshould support stricter enforcement for cases where there is a well-understood set of valid options, but this should be fine for now.  Most likely we'd need to add another flag to the param to set the enforcement level, and/or a separate set of deprecated values that are accepted but trigger the log message.

Reply:
Ok, but at that point we can't be certain if the given value is indeed a deprecated value or if it's a completely invalid value. So if the message says it's deprecated, that might also lead to confusing situations or not? How about:

```
Invalid or deprecated value "{value}" specified for parameter "{param}" of function "{func}"'
```